independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/workspace/WorkspaceModule.java

```diff
@@ -12,13 +14,34 @@
 
     File getBuildDir();
 
-    Collection<ProcessedSources> getMainSources();
+    Collection<String> getSourceClassifiers();
 
-    Collection<ProcessedSources> getMainResources();
+    boolean hasSources(String classifier);
 
-    Collection<ProcessedSources> getTestSources();
+    ArtifactSources getSources(String classifier);
 
-    Collection<ProcessedSources> getTestResources();
+    default boolean hasMainSources() {
+        return hasSources(DefaultWorkspaceModule.MAIN);
+    }
+
+    default boolean hasTestSources() {
+        return hasSources(DefaultWorkspaceModule.TEST);
+    }
+
+    default ArtifactSources getMainSources() {
+        return getSources(DefaultWorkspaceModule.MAIN);
+    }
+
+    default ArtifactSources getTestSources() {
+        return getSources(DefaultWorkspaceModule.TEST);
+    }
 
     PathCollection getBuildFiles();
+
+    default PathTree getContentTree(String classifier) {
+        //System.out.println("WorkspaceModule.getContentTree " + /* getId() + */ ":" + classifier);
```

Comment:
This can probably be removed now?

Reply:
Yes, I'll remove it, I kept it as a reminder this line could break dev mode.